src/routes/delegates/delegates.service.ts

```diff
@@ -87,4 +94,21 @@ export class DelegatesService {
       deleteDelegateDto.signature,
     );
   }
+
+  async deleteSafeDelegate(
+    chainId: string,
+    delegateAddress: string,
+    deleteSafeDelegateRequest: DeleteSafeDelegateDto,
+  ): Promise<unknown> {
+    if (delegateAddress !== deleteSafeDelegateRequest.delegate) {
+      throw new BadRequestException();
+    }
```

Comment:
Why do we repeat the check here?

Reply:
This one is for checking the delegate address, the one in the controller is checking the safe address, but anyway, it makes no sense to have validation in both places, I'll change it 👍

Reply:
Removed in [74dde61](https://github.com/5afe/safe-client-gateway-nest/pull/270/commits/74dde610ba8bbbdcb48c03380f2c23a1a6c9f9fc)